### PR TITLE
[lib] [mutex] Fix build for OCaml 5.x due to missing caml_mutex_unlock

### PR DESCRIPTION
**Pull request by ejgallego**
Problem introduced in #19175

### CODE REVIEW

**Comment by ejgallego** [comment]
CI with OCaml 5.x series at https://github.com/ejgallego/coq-lsp/pull/793/checks

**Review by SkySkimmer — APPROVED** [review]

**Comment by SkySkimmer** [comment]
@coqbot run full ci

**Comment by ejgallego** [comment]
`coq-lsp` CI passed for all relevant OCaml versions.

File: clib/memprof_coq.memprof.ml [hunk 1]
```diff
@@ -8,23 +8,8 @@ module Resource_bind = Memprof_limits.Resource_bind
...
-     - Inline the call to Mutex.unlock to avoid polling in bytecode.
-       (workaround to the lack of masking) *)
-  let[@inline never] with_lock m ~scope =
-    let () = Mutex.lock m (* BEGIN ATOMIC *) in
-    match (* END ATOMIC *) scope () with
-    | (* BEGIN ATOMIC *) x -> unlock m ; (* END ATOMIC *) x
-    | (* BEGIN ATOMIC *) exception e -> unlock m ; (* END ATOMIC *) raise e
-
-end
+(* We do own own Mutex_aux for OCaml 5.x *)
```
Comment (gadmm) [inline clib/memprof_coq.memprof.ml:11]:
```suggestion
(* We do our own Mutex_aux for OCaml 5.x *)
```

File: clib/mutex_aux_5.0.ml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+(* EJGA: probably unsafe, but OCaml 5.0 is just not good for us, so we
+   add this code as to enable compilation *)
+let unlock = Mutex.unlock
```
Comment (gadmm) [inline clib/mutex_aux_5.0.ml:3]:
```suggestion
(* backport of Mutex.protect from OCaml 5.1 *)
external unlock: t -> unit = "caml_ml_mutex_unlock"
```

File: clib/memprof_coq.std.ml [hunk 3]
```diff
@@ -4,23 +4,8 @@ module Resource_bind = struct
...
-     - Inline the call to Mutex.unlock to avoid polling in bytecode.
-       (workaround to the lack of masking) *)
-  let[@inline never] with_lock m ~scope =
-    let () = Mutex.lock m (* BEGIN ATOMIC *) in
-    match (* END ATOMIC *) scope () with
-    | (* BEGIN ATOMIC *) x -> unlock m ; (* END ATOMIC *) x
-    | (* BEGIN ATOMIC *) exception e -> unlock m ; (* END ATOMIC *) raise e
-
-end
+(* We do own own Mutex_aux for OCaml 5.x *)
```
Comment (gadmm) [inline clib/memprof_coq.std.ml:7]:
```suggestion
(* We do our own Mutex_aux for OCaml 5.x *)
```

**Comment by ejgallego** [comment]
Changes added, thanks @gadmm !

New OCaml 5.x CI at https://github.com/ejgallego/coq-lsp/pull/793/checks

**Comment by SkySkimmer** [comment]
@coqbot merge now

**Pull request merged**